src/trees/FunctionTree.cpp

```diff
@@ -389,6 +389,34 @@ template <int D> void FunctionTree<D>::multiply(double c, FunctionTree<D> &inp)
     inp.deleteGenerated();
 }
 
+/** @brief In-place mapping with a predefined function f(x), fixed grid
+ *
+ * @param[in] fmap: mapping function
+ *
+ * @details The input function will be mapped in-place on the current grid
+ * of the function, i.e. no further grid refinement.
+ *
+ */
+template <int D> void FunctionTree<D>::map(std::function<double(double)> fmap) {
+    if (this->getNGenNodes() != 0) MSG_ABORT("GenNodes not cleared");
+    {
+        int nNodes = this->getNEndNodes();
+#pragma omp for schedule(guided)
```

Comment:
What about using a range-based `for` loop?
```c++
for (auto & node: this->endNodeTable) {
      node.mwTransform(Reconstruction);
      node.cvTransform(Forward);
     for (auto & coef : node.getCoefs()) {
            coef = fmap(coef);
      }
      node.cvTransform(Backward);
      node.mwTransform(Compression);
      node.calcNorms();
}
```
I am unsure about the OpenMP `pragma`-s play out with this though, @stigrj opinions?

Reply:
This should be possible with newer versions of OpenMP. I did a quick test and it works on my laptop, but not on Saga and Travis, so let's just keep it as is for now

Reply:
Bummer.

Reply:
I have copied the function from the in-place multiplication, and the `#pragma` stems from that one. I will let @stigrj evaluate if that should stay there or not.